### PR TITLE
Introduce a dts for nrf52_bsim

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
@aescolar my first attempt at introduce a board dts for nrf52_bsim (makes life a lot easier going forward).  Not sure what else we might want to "delete" from the SoC dts.

File: scripts/dts/gen_defines.py [hunk 1]
```diff
@@ -505,7 +505,10 @@ def write_chosen(edt):
     for name, node in edt.chosen_nodes.items():
         chosen[f"DT_CHOSEN_{str2ident(name)}"] = f"DT_{node.z_path_id}"
         chosen[f"DT_CHOSEN_{str2ident(name)}_EXISTS"] = 1
-    max_len = max(map(len, chosen))
+    if len(chosen):
+        max_len = max(map(len, chosen))
```
Comment (mbolivar-nordic) [inline scripts/dts/gen_defines.py]: can just write:

```
max_len = max(map(len, chosen), default=0)
```

**Review by mbolivar-nordic — COMMENTED** [review]
python parts look good

**Comment by galak** [comment]
@mb

> python parts look good



> python parts look good

I pulled the python bit into https://github.com/zephyrproject-rtos/zephyr/pull/23791

**Comment by carlescufi** [comment]
> @aescolar my first attempt at introduce a board dts for nrf52_bsim (makes life a lot easier going forward). Not sure what else we might want to "delete" from the SoC dts.

The nrf52_bsim supports a very small  subset of the nRF52 peripherals. As far as I know it mostly supports the following peripherals:
- `TIMER`
- `CLOCK`
- `POWER`
- `PPI`
- `FICR`
- `RTC`
- `RADIO`
- `RNG`
- `ECB`
- `CCM`
- `AAR` 

So anything else can be deleted. In this case I would assume the following can be safely removed:

- `i2c`
- `spi`
- `uart`
- `adc`
- `gpio`
- `gpiote`
- `wdt`
- `pwm`
- `qdec`
- `sw_pwm`

**Comment by wopu-ot** [comment]
Is the goal to have the board dts match what nrf52_bsim supports, or just to get rid of the hack?

**Comment by galak** [comment]
> Is the goal to have the board dts match what nrf52_bsim supports, or just to get rid of the hack?

we'll the goal is to get rid of the hack.  If we can match what's supported that would also be good.

**Comment by galak** [comment]
Updated dts to remove devices not know to be simulated.

**Review by wopu-ot — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]

**Pull request merged**